### ISSUE TITLE
something wrong with tqdm module

### ISSUE THREAD

**Issue by wulidongdong** [nteract/hydrogen#1293]
hi, devs:
After i update my hydrogen to v2.4.0, the tqdm module doesnt work well. The progress bar does not change during the iteration, only suddenly jump to 100% after iteration done. it works well with hydrogen2.3.0, and my atom crashed couple of times after hydrogen update.

OS: windows10
atom:1.25.1
hydrogen:2.4.0
tqdm:4.22.0
python:3.6.4

thanks.

**Comment by lgeiger** [comment]
I guess this is related to our recent MobX upgrade in #1284

**Comment by BenRussert** [comment]
I agree @lgeiger . @wulidongdong do you have an example code snippet i can use to reproduce?

**Comment by BenRussert** [comment]
I modified a simple example from the `tqdm` docs, adding a sleep call to make it take some time and show progress:

```python
import time
from tqdm import tqdm

for i in tqdm(range(10)):
    time.sleep(1)
    pass

```


Still looking into this, but I've noticed it works in the output area dock, but in regular inline result views.

**Comment by AstrickHarren** [comment]
Hey devs:
Somethings happening again like this. 
The example is not working well:
```python
import time

for i in range(10):
    time.sleep(1)
    print(i)

Regular inline result view or dock view are both not behaving well: they all give the first 1 or 2 output then wait another 8 secs and give out all the output at once. Same thing happens with `tqdm`.

OS: MacOS Mojave 10.14.2
atom: 1.38.2
python: 3.7.0
tqdm: 4.28.1
hydrogen: 2.10.0

thx.

**Comment by jacquespeeters** [comment]
Not sure if you are aware that there are some trouble with Hydrogen Output area and tqdm.

I struggled for a long time to make tqdm work with Hydrogen Output Area. I the following problem when using Hydrogen, interupting tqdm, debugging, then running tqdm loop back.

![image](https://user-images.githubusercontent.com/15957925/77262141-161a3f80-6c94-11ea-8aa5-a05d05aad2e9.png)

From my understanding, it can be solve on Jupyter notebooks by using tqdm_notebook but it doens't work on Hydrogen (it relies on HTML widgets).

Was able to solve my problem by following this addice.
https://stackoverflow.com/a/60067890/5247364

However it throws and error if no instances exists. If you still want to use it with scripts and Hydrogen IDE. Here is my code. 

```
try:
    # Avoids problem on notebooks / Hydrogen IDE
    tqdm.tqdm._instances.clear()
except Exception:
    pass

# i advise using sys.stdout with Hydrogen Output Area
for my_element in tqdm.tqdm(my_list, file=sys.stdout):
  # do something
```

Sorry if I'm off-topic but I did struggle a lot to make it work. Google queries send it back to here, so it might help others.
